lib/gears/shape.lua

```diff
@@ -87,21 +97,41 @@ function module.infobubble(cr, width, height, corner_radius, arrow_size, arrow_p
     cr:close_path()
 end
 
---- A rectangle terminated by an arrow
+--- A rectangle terminated by an arrow.
+--
+-- ![Shape example](@SHAPE_rectangular_tag_SVG@)
+--
+-- @SHAPE_rectangular_tag_EXAMPLE@
+--
 -- @param cr A cairo context
 -- @tparam number width The shape with
 -- @tparam number height The shape height
-function module.rectangular_tag(cr, width, height)
-    cr:move_to(0        , height/2)
-    cr:line_to(height/2 , 0       )
-    cr:line_to(width    , 0       )
-    cr:line_to(width    , height  )
-    cr:line_to(height/2 , height  )
+-- @tparam[opt=height/2] number arrow_length The length of the arrow part
+function module.rectangular_tag(cr, width, height, arrow_length)
+    local arrow_length = arrow_length or height/2
+    if arrow_length > 0 then
+        cr:move_to(0            , height/2 )
+        cr:line_to(arrow_length , 0        )
+        cr:line_to(width        , 0        )
+        cr:line_to(width        , height   )
+        cr:line_to(arrow_length , height   )
+    else
+        cr:move_to(0            , 0        )
+        cr:line_to(-arrow_length, height/2 )
+        cr:line_to(0            , height   )
+        cr:line_to(width        , height   )
+        cr:line_to(width        , 0        )
+    end
```

Comment:
Do I understand it correctly that a negative `arrow_length` makes the arrow point to the other direction? If so, should that be mentioned in the docs?